设置了 PUID PGID 后文件所有者仍然为root

我的启动命令如下
```
docker run -d \
  --name cloudreve \
  -e PUID=1000 \
  -e PGID=100 \
  -e TZ="Asia/Shanghai" \
  -p 5212:5212 \
  --restart=unless-stopped \
  -v /home/sherry/cloudreve/uploads:/cloudreve/uploads \
  -v /home/sherry/cloudreve/config:/cloudreve/config \
  -v /home/sherry/cloudreve/db:/cloudreve/db \
  -v /home/sherry/cloudreve/avatar:/cloudreve/avatar \
  xavierniu/cloudreve:latest
```
启动后，几个文件夹的所有者为PUID=1000的用户，但是文件夹中的文件所有者还是为root用户

使用user后，如果为非root用户，会出现权限问题

日志如下

```
[Panic]   2021-08-29 16:59:02 无法创建配置文件, open /cloudreve/config/conf.ini: permission denied
panic: 无法创建配置文件, open /cloudreve/config/conf.ini: permission denied

goroutine 1 [running]:
github.com/cloudreve/Cloudreve/v3/pkg/util.(*Logger).Panic(0xc0000c21e0, 0x11eb2ec, 0x1c, 0xc00021fab0, 0x1, 0x1)
        /ProjectCloudreve/Cloudreve/pkg/util/logger.go:74 +0xe5
github.com/cloudreve/Cloudreve/v3/pkg/conf.Init(0x7fffd3dfdf15, 0x1a)
        /ProjectCloudreve/Cloudreve/pkg/conf/conf.go:108 +0xc36
github.com/cloudreve/Cloudreve/v3/bootstrap.Init(0x7fffd3dfdf15, 0x1a)
        /ProjectCloudreve/Cloudreve/bootstrap/init.go:18 +0x3e
main.init.0()
        /ProjectCloudreve/Cloudreve/main.go:23 +0x205
```

我的启动命令如下
```
docker run -d \
  --name cloudreve \
  -e TZ="Asia/Shanghai" \
  -p 5212:5212 \
  --user 1000:100 \
  --restart=unless-stopped \
  -v /home/sherry/cloudreve/uploads:/cloudreve/uploads \
  -v /home/sherry/cloudreve/config:/cloudreve/config \
  -v /home/sherry/cloudreve/db:/cloudreve/db \
  -v /home/sherry/cloudreve/avatar:/cloudreve/avatar \
  xavierniu/cloudreve:latest
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -33,7 +33,8 @@ WORKDIR /cloudreve
 COPY --from=builder /ProjectCloudreve/Cloudreve/cloudreve-main /cloudreve/
 
 RUN echo ">>>>>> set up PUID and PGID" \
-    && useradd appuser -r -u $PUID -g $PGID
+    && addgroup -S -g $PGID appuser \
+    && adduser -S -u $PUID -s /sbin/nologin -G appuser appuser
 
 USER appuser
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -28,15 +28,9 @@ ENV TZ="Asia/Shanghai"
 
 LABEL MAINTAINER="Xavier Niu"
 
-WORKDIR /cloudreve
+WORKDIR /Cloudreve
 
-COPY --from=builder /ProjectCloudreve/Cloudreve/cloudreve-main /cloudreve/
-
-RUN echo ">>>>>> set up PUID and PGID" \
-    && addgroup -S -g $PGID appuser \
-    && adduser -S -u $PUID -s /sbin/nologin -G appuser appuser
-
-USER appuser
+COPY --from=builder /ProjectCloudreve/Cloudreve/cloudreve-main ./
 
 RUN echo ">>>>>> update dependencies" \
     && apk update \
@@ -44,8 +38,14 @@ RUN echo ">>>>>> update dependencies" \
     && echo ">>>>>> set up timezone" \
     && cp /usr/share/zoneinfo/${TZ} /etc/localtime \
     && echo ${TZ} > /etc/timezone \
+    && echo ">>>>>> set up PUID and PGID" \
+    && addgroup -S -g $PGID appuser \
+    && adduser -S -u $PUID -s /sbin/nologin -G appuser appuser \
     && echo ">>>>>> fix cloudreve-main premission" \
-    && chmod +x /cloudreve/cloudreve-main
+    && chmod 740 cloudreve-main \
+    && chown appuser:appuser cloudreve-main
+
+USER appuser
 
 VOLUME ["/cloudreve/uploads", "/downloads", "/cloudreve/avatar", "/cloudreve/config", "/cloudreve/db"]
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,7 +20,7 @@ RUN cd ./Cloudreve \
     && export COMMIT_SHA=$(git rev-parse --short HEAD) \
     && go build -a -o cloudreve-main -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$CLOUDREVE_VERSION' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
 
-FROM lsiobase/alpine:3.13
+FROM alpine:3.13
 
 ENV PUID=1000
 ENV PGID=1000
@@ -32,7 +32,10 @@ WORKDIR /cloudreve
 
 COPY --from=builder /ProjectCloudreve/Cloudreve/cloudreve-main /cloudreve/
 
-VOLUME ["/cloudreve/uploads", "/downloads", "/cloudreve/avatar", "/cloudreve/config", "/cloudreve/db"]
+RUN echo ">>>>>> set up PUID and PGID" \
+    && useradd appuser -r -u $PUID -g $PGID
+
+USER appuser
 
 RUN echo ">>>>>> update dependencies" \
     && apk update \
@@ -42,6 +45,8 @@ RUN echo ">>>>>> update dependencies" \
     && echo ${TZ} > /etc/timezone \
     && echo ">>>>>> fix cloudreve-main premission" \
     && chmod +x /cloudreve/cloudreve-main
+
+VOLUME ["/cloudreve/uploads", "/downloads", "/cloudreve/avatar", "/cloudreve/config", "/cloudreve/db"]
 
 EXPOSE 5212
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,7 +20,7 @@ RUN cd ./Cloudreve \
     && export COMMIT_SHA=$(git rev-parse --short HEAD) \
     && go build -a -o cloudreve-main -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$CLOUDREVE_VERSION' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
 
-FROM ghcr.io/linuxserver/baseimage-alpine:3.14
+FROM alpine:3.13
 
 ENV TZ="Asia/Shanghai"
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -22,8 +22,6 @@ RUN cd ./Cloudreve \
 
 FROM alpine:3.13
 
-ENV PUID=1000
-ENV PGID=1000
 ENV TZ="Asia/Shanghai"
 
 LABEL MAINTAINER="Xavier Niu"
@@ -38,14 +36,8 @@ RUN echo ">>>>>> update dependencies" \
     && echo ">>>>>> set up timezone" \
     && cp /usr/share/zoneinfo/${TZ} /etc/localtime \
     && echo ${TZ} > /etc/timezone \
-    && echo ">>>>>> set up PUID and PGID" \
-    && addgroup -S -g $PGID appuser \
-    && adduser -S -u $PUID -s /sbin/nologin -G appuser appuser \
     && echo ">>>>>> fix cloudreve-main premission" \
-    && chmod 740 cloudreve-main \
-    && chown appuser:appuser cloudreve-main
-
-USER appuser
+    && chmod +x cloudreve-main
 
 VOLUME ["/cloudreve/uploads", "/downloads", "/cloudreve/avatar", "/cloudreve/config", "/cloudreve/db"]
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,7 +20,7 @@ RUN cd ./Cloudreve \
     && export COMMIT_SHA=$(git rev-parse --short HEAD) \
     && go build -a -o cloudreve-main -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$CLOUDREVE_VERSION' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
 
-FROM alpine:3.13
+FROM ghcr.io/linuxserver/baseimage-alpine:3.14
 
 ENV TZ="Asia/Shanghai"
 

--- a/README-DOCKER-COMPOSE.md
+++ b/README-DOCKER-COMPOSE.md
@@ -33,6 +33,24 @@ server {
 }
 ```
 
+目录
+
+- `<PATH TO config>`: 配置文件夹，如`/dockercnf/cloudreve/config`
+
+创建配置文件夹
+
+```bash
+mkdir -p <PATH TO config>
+```
+
+创建配置文件`vim <PATH TO config>/conf.ini `（*该配置文件针对SQLite数据库，如需使用MySQL等数据库，请参见cloudreve官方文档*）
+
+```ini
+# conf.ini
+[Database]
+DBFile = /cloudreve/db/cloudreve.db
+```
+
 ### 下载环境文件以及Docker Compose文件
 
 下载环境文件

--- a/README-NAC.md
+++ b/README-NAC.md
@@ -74,11 +74,32 @@ docker run -d \
 
 ### 启动Cloudreve
 
+目录
+
+- `<PATH TO uploads>`:上传目录，如`/sharedfolders`
+- `<PATH TO config>`: 配置文件夹，如`/dockercnf/cloudreve/config`
+- `<PATH TO db>`: 数据库文件夹，如`/dockercnf/cloudreve/db`
+- `<PATH TO avatar>`: 头像文件夹，如`/dockercnf/cloudreve/avatar`
+
+创建配置文件夹
+
+```bash
+mkdir -p <PATH TO config>
+```
+
+创建配置文件`vim <PATH TO config>/conf.ini `（*该配置文件针对SQLite数据库，如需使用MySQL等数据库，请参见cloudreve官方文档*）
+
+```ini
+# conf.ini
+[Database]
+DBFile = /cloudreve/db/cloudreve.db
+```
+
+启动cloudreve容器
+
 ```bash
 docker run -d \
   --name cloudreve \
-  -e PUID=1000 \ # optional
-  -e PGID=1000 \ # optional
   -e TZ="Asia/Shanghai" \ # optional
   --network my-network \
   --restart=unless-stopped \
@@ -93,14 +114,7 @@ docker run -d \
 说明
 
 - 首次启动后请执行`docker logs -f cloudreve`获取初始密码
-
-- PUID以及PGID的获取方式详见`获取PUID和PGID`
-
-- `<PATH TO uploads>`:上传目录, 例如`/sharedfolders`
-- `<PATH TO temp>`: 临时下载文件夹，需要与Aria的`/downloads`对应，例如`/dockercnf/aria2/temp`（如不需要离线下载功能`#1`可以删除）
-- `<PATH TO config>`: 配置文件夹，如`/dockercnf/cloudreve/config`
-- `<PATH TO db>`: 数据库文件夹，如`/dockercnf/cloudreve/db`
-- `<PATH TO avatar>`: 头像文件夹，如`/dockercnf/cloudreve/avatar`
+- `TZ`设置时区，默认值为`Asia/Shanghai`
 
 ### 配置Cloudreve连接Aria2服务器
 

--- a/README.md
+++ b/README.md
@@ -12,16 +12,6 @@
 - 简易安装
 - 内含详细的Cloudreve+Nginx+Aria2部署教程
 
-## 获取PUID和PGID
-
-为什么要使用PUID和PGID参见[Understanding PUID and PGID](https://docs.linuxserver.io/general/understanding-puid-and-pgid)。假设当前登陆用户为`root`，则执行`id root`就会得到类似于下面的一段代码：
-
-```
-uid=1000(root) gid=1001(root)
-```
-
-则在运行命令中的PUID填入`1000`，PGID填入`1001`。
-
 ## 开始
 
 目录
@@ -50,8 +40,6 @@ DBFile = /cloudreve/db/cloudreve.db
 ```bash
 docker run -d \
   --name cloudreve \
-  -e PUID=1000 \ # optional
-  -e PGID=1000 \ # optional
   -e TZ="Asia/Shanghai" \ # optional
   -p 5212:5212 \
   --restart=unless-stopped \
@@ -65,7 +53,6 @@ docker run -d \
 说明
 
 - 首次启动后请执行`docker logs -f cloudreve`获取初始密码
-- PUID以及PGID的获取方式详见`获取PUID和PGID`
 - `TZ`设置时区，默认值为`Asia/Shanghai`
 
 其他教程

